crates/trie/src/updates.rs

```diff
@@ -136,21 +134,19 @@ impl TrieUpdates {
                     TrieOp::Update(..) => unreachable!("Cannot update full storage trie."),
                 },
                 TrieKey::StorageNode(hashed_address, nibbles) => {
-                    if !nibbles.inner.is_empty() {
-                        // Delete the old entry if it exists.
-                        if storage_trie_cursor
-                            .seek_by_key_subkey(hashed_address, nibbles.clone())?
-                            .filter(|e| e.nibbles == nibbles)
-                            .is_some()
-                        {
-                            storage_trie_cursor.delete_current()?;
-                        }
+                    // Delete the old entry if it exists.
```

Comment:
The original code disallow root node to store in db. I think the reason is the `TrieWalker` api only allow set `retain_updates` after `TrieWalker::new` like `TrieWalker::new().with_updates(retain_updates)`, but it is too late, since the root node has already loaded in `new()` so `TrieUpdates::schedule_delete` is not called for the root node.